extensions/amp-sidebar/1.0/component.js

```diff
@@ -135,42 +136,110 @@ function SidebarWithRef(
-          hidden={!side}
-          {...rest}
-        >
-          {children}
-        </ContainWrapper>
-        <div
-          ref={backdropRef}
-          onClick={() => close()}
-          part="backdrop"
-          style={backdropStyle}
-          className={`${backdropClassName ?? ''} ${classes.backdrop} ${
-            classes.defaultBackdropStyles
-          }`}
-          hidden={!side}
-        >
-          <div className={classes.backdropOverscrollBlocker}></div>
-        </div>
-      </>
-    )
+    <div className={mounted ? '' : classes.unmounted}>
+      <ContainWrapper
+        as={Comp}
+        ref={sidebarRef}
+        size={false}
+        layout={true}
+        paint={true}
+        part="sidebar"
+        wrapperClassName={`${classes.sidebar} ${classes.defaultSidebarStyles} ${
+          side === Side.LEFT ? classes.left : classes.right
+        }`}
+        role="menu"
+        tabindex="-1"
+        hidden={!side}
+        {...rest}
+      >
+        {children}
+      </ContainWrapper>
+      <div
+        ref={backdropRef}
+        onClick={() => close()}
+        part="backdrop"
+        style={backdropStyle}
+        className={`${backdropClassName ?? ''} ${classes.backdrop} ${
+          classes.defaultBackdropStyles
+        }`}
+        hidden={!side}
+      >
+        <div className={classes.backdropOverscrollBlocker}></div>
+      </div>
+    </div>
   );
 }
 
 const Sidebar = forwardRef(SidebarWithRef);
 Sidebar.displayName = 'Sidebar'; // Make findable for tests.
 export {Sidebar};
+
+/**
+ * @param {!SidebarDef.SidebarToolbar} props
+ * @return {PreactDef.Renderable}
+ */
+export function SidebarToolbar({
+  toolbar,
+  'toolbar-target': toolbarTarget,
+  children,
+  ...rest
+}) {
+  const ref = useRef();
+  const [targetEl, setTargetEl] = useState();
+  const [, setReRender] = useState(true);
+
+  // Trigger re-render each time the window is resized
+  useEffect(() => {
+    const navElement = ref.current;
+    if (!navElement) {
+      return;
+    }
+    const document = navElement.ownerDocument;
+    if (!document) {
+      return;
+    }
+    const window = document.defaultView;
+    if (!window) {
+      return;
+    }
+
+    const reRenderFunction = debounce(
+      window,
+      () => setReRender((prev) => !prev),
+      100
+    );
+    window.addEventListener('resize', reRenderFunction);
+    return () => {
+      window.removeEventListener('resize', reRenderFunction);
+    };
+  }, []);
+
+  useEffect(() => {
+    const navElement = ref.current;
+    if (!navElement) {
+      return;
+    }
+    const document = navElement.ownerDocument;
+    if (!document) {
+      return;
+    }
+    const window = document.defaultView;
+    if (!window) {
+      return;
+    }
+
+    const matches = window.matchMedia(toolbar).matches;
+    const selector = `#${CSS.escape(toolbarTarget)}`;
+    const newTargetEl = document.querySelector(selector);
+    setTargetEl(matches && newTargetEl);
```

Comment:
Let's narrow the type to `Element | null` instead of `Element | boolean | null`
```suggestion
    setTargetEl(matches ? newTargetEl : null);
```

Reply:
Thanks!  Updated in the latest design!